### PR TITLE
Implement support for ARM64_32. Fixes #4864.

### LINKED ISSUE #4864: [watchos] Can't deploy with ARMv7k architecture on Series 4
### Steps to Reproduce

1. Create a new watchOS project.
2. Deploy on new Apple Watch Series 4.

### Expected Behavior

Should deploy fine.

### Actual Behavior

Deploying to Device output:

```
IncorrectArchitecture: Failed to find matching arch for 32-bit Mach-O input file /private/var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.mC7GED/extracted/MySingleAppMyWatchProject.app/PlugIns/MySingleAppMyWatchProjectExtension.appex/MySingleAppMyWatchProjectExtension
error MT1006: Could not install the application '/Users/vince/Projects/MySingleApp/MySingleApp/bin/iPhone/Debug/device-builds/iphone10.6-12.1/MySingleApp.app' on the device 'Vincent’s iPhone X': AMDeviceSecureInstallApplicationBundle returned: 0xe8000087 (kAMDIncorrectArchitectureError).
```

### Environment

```
=== Visual Studio Enterprise 2017 for Mac ===

Version 7.6.6 (build 17)
Installation UUID: bef2b48b-cfbf-4371-bca9-fa440adab7db
Runtime:
	Mono 5.12.0.301 (2018-02/4fe3280bba1) (64-bit)
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 4.4.1.178 (master / eeaeb7e6)

	Package version: 512000301

=== NuGet ===

Version: 4.3.1.4445

=== .NET Core ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	2.1.2
	2.0.5
	2.0.0
SDK: /usr/local/share/dotnet/sdk/2.1.302/Sdks
SDK Versions:
	2.1.302
	2.1.4
	2.0.2
	2.0.0
MSBuild SDKs: /Library/Frameworks/Mono.framework/Versions/5.12.0/lib/mono/msbuild/15.0/bin/Sdks

=== Xamarin.Profiler ===

Version: 1.6.3
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Apple Developer Tools ===

Xcode 10.1 (14460.20.3)
Build 10O23u

=== Xamarin.Android ===

Version: 9.0.0.20 (Visual Studio Enterprise)
Android SDK: /Users/vince/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		4.1    (API level 16)
		4.2    (API level 17)
		4.3    (API level 18)
		4.4    (API level 19)
		4.4.87 (API level 20)
		5.0    (API level 21)
		5.1    (API level 22)
		6.0    (API level 23)
		7.0    (API level 24)
		7.1    (API level 25)
		8.0    (API level 26)
		8.1    (API level 27)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 26.0.2
SDK Build Tools Version: 27.0.0

Java SDK: /usr
java version "1.8.0_101"
Java(TM) SE Runtime Environment (build 1.8.0_101-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.101-b13, mixed mode)

Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Xamarin.Mac ===

Version: 4.6.0.14 (Visual Studio Enterprise)
Hash: 1a6e2786
Branch: 
Build date: 2018-08-28 00:12:06-0400

=== Xamarin.iOS ===

Version: 12.0.0.15 (Visual Studio Enterprise)
Hash: 84552a46
Branch: xcode10
Build date: 2018-09-17 21:54:33-0400

=== Xamarin Inspector ===

Version: 1.4.3
Hash: db27525
Branch: 1.4-release
Build date: Mon, 09 Jul 2018 21:20:18 GMT
Client compatibility: 1

=== Build Information ===

Release ID: 706060017
Git revision: 822323b5f6295b271d4c9bfaca4129f90ec41f38
Build date: 2018-09-18 20:19:40+00
Build branch: release-7.6-xcode10
Xamarin extensions: a83eb4b7ad9174ac2e705f2c4257ecd15688ba97

=== Operating System ===

Mac OS X 10.14.0
Darwin 18.0.0 Darwin Kernel Version 18.0.0
    Wed Aug 22 20:59:21 PDT 2018
    root:xnu-4903.201.2~4/RELEASE_X86_64 x86_64

=== Enabled user installed extensions ===

NuGet Package Explorer 0.2
Internet of Things (IoT) development (Preview) 7.5

```

### Build Logs

https://gist.github.com/VincentDondain/574b4fad1933e6a0c20b8df0c8a7ade9

> VS bug [#891355](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/891355)

### PR DESCRIPTION
**Pull request by rolfbjarne**
This PR is best reviewed commit-by-commit.

Fixes https://github.com/xamarin/xamarin-macios/issues/4864.

### CODE REVIEW

File: tests/introspection/iOS/iOSApiCtorInitTest.cs [hunk 1]
```diff
@@ -225,6 +225,11 @@ protected override bool Skip (Type type)
 			case "INGetRestaurantGuestIntentResponse": // Objective-C exception thrown.  Name: NSInternalInconsistencyException Reason: Unable to initialize 'INGetRestaurantGuestIntentResponse'. Please make sure that your intent definition file is valid.
 				return TestRuntime.CheckXcodeVersion (10,0);
 			case "CMMovementDisorderManager": // Not available in simulator, added info to radar://41110708 
+#if __WATCHOS__
+                    // Doesn't exist in the simulator; aborts on device if the required entitlement isn't available.
+                    return true;
+#endif
+                    return Runtime.Arch == Arch.SIMULATOR;
```
Comment (spouliot) [inline tests/introspection/iOS/iOSApiCtorInitTest.cs]: minor: alignment

**Comment by spouliot** [comment]
* Enable the varargs dynamic trampolines for arm64_32.

Mentioned as not fully functional. Is there an open issue for it ?

**Comment by rolfbjarne** [comment]
> * Enable the varargs dynamic trampolines for arm64_32.
> 
> Mentioned as not fully functional. Is there an open issue for it ?

https://github.com/xamarin/xamarin-macios/issues/6041

**Comment by rolfbjarne** [comment]
Test failures are unrelated

* MSBuild tests/iOS: https://github.com/xamarin/maccore/issues/1601
* MTouch tests: https://github.com/xamarin/maccore/issues/1601
* Mono BCL tests group 5/iOS Unified 32-bits: https://github.com/xamarin/maccore/issues/1632
* System.Security/watchOS: https://github.com/xamarin/maccore/issues/581

**Comment by rolfbjarne** [comment]
@monojenkins backport d16-2

**Pull request merged**